docs/docs/development/hardware-integration/physical-layouts.md

```diff
@@ -0,0 +1,351 @@
+
+#### Example larger position map
+
+Consider the following macropad/numpad with two physical layouts:
+
+![A 4x5 numpad/macropad](../../assets/hardware-integration/numpad.png)
+
+Let us first consider each side individually. The "reference" position map of the left side would look like this:
+
+```dts
+macropad {
+    physical-layout = <&macropad_layout>;
+    positions
+        = < 0  1  2  3>
+        , < 4  5  6  7>
+        , < 8  9 10 11>
+        , <12 13 14 15>
+        , <16 17 18 19>;
+};
+```
+
+Meanwhile, the "reference" position map of the right side with fewer keys would look like this:
+
+```dts
+numpad {
+    physical-layout = <&numpad_layout>;
+    positions
+        = < 0  1  2  3>
+        , < 4  5  6  7>
+        , < 8  9 10   >
+        , <11 12 13 14>
+        , <15    16   >;
+};
+```
+
+As a reminder, the ordering of the physical layout and the position map is the same as in the matrix transform and the keymap (all are just one-dimensional arrays presented nicely through the use of whitespace).
+
+If the left side with more keys was used as the reference layout, then the overall position map of the keyboard would look like this:
+
+```dts
+/ {
+    keypad_lossless_position_map {
+        compatible = "zmk,physical-layout-position-map";
+        complete;
+
+        macropad {
+            physical-layout = <&macropad_layout>;
+            positions
+                = < 0  1  2  3>
+                , < 4  5  6  7>
+                , < 8  9 10 11>
+                , <12 13 14 15>
+                , <16 17 18 19>;
+        };
+
+        numpad {
+            physical-layout = <&numpad_layout>;
+            positions
+                = < 0  1  2  3>
+                , < 4  5  6  7>
+                , < 8  9 10 17>
+                , <11 12 13 14>
+                , <15 18 16 19>;
+        };
+    };
+};
+```
+
+The "missing" positions are filled with the "spare" numbers of the layout with more keys. Meanwhile, if the right side with fewer keys is used as a reference, then the overall position map would look like this:
+
+```dts
+/ {
+    keypad_lossy_position_map {
+        compatible = "zmk,physical-layout-position-map";
+        complete;
+
+        macropad {
+            physical-layout = <&macropad_layout>;
+            positions
+                = < 0  1  2  3>
+                , < 4  5  6  7>
+                , < 8  9 10   >
+                , <12 13 14 15>
+                , <16    18   >;
+        };
+
+        numpad {
+            physical-layout = <&numpad_layout>;
+            positions
+                = < 0  1  2  3>
+                , < 4  5  6  7>
+                , < 8  9 10   >
+                , <11 12 13 14>
+                , <15    16   >;
+        };
+    };
+};
+```
+
```

Comment:
Is it worth adding a note here on why the former is preferable to this? i.e. that the extra positions can be recovered

Reply:
Added a note.